File: examples/android/demo/MoQDemo/app/src/main/java/com/swmansion/moqdemo/features/chat/ChatModels.kt
package com.swmansion.moqdemo.features.chat

import java.util.UUID

data class ChatPayload(
    val from: String,
    val message: String,
)

data class ChatMessage(
    val direction: Direction,
    val from: String,
    val text: String,
    val broadcastPath: String,
    val timestampMs: Long = System.currentTimeMillis(),
    val id: String = UUID.randomUUID().toString(),
) {
    enum class Direction {
        Local,
        Remote,
    }

    val isLocal: Boolean
        get() = direction == Direction.Local
}

File: examples/ios/demo/MoQDemo/Features/Chat/Models/ChatModels.swift
import Foundation

struct ChatPayload: Codable, Equatable {
    let from: String
    let message: String
}

struct ChatMessage: Identifiable, Equatable {
    enum Direction: Equatable {
        case local
        case remote
    }

    let id = UUID()
    let direction: Direction
    let from: String
    let text: String
    let broadcastPath: String
    let timestamp: Date

    var isLocal: Bool {
        direction == .local
    }
}

